### PR TITLE
[HEI-258] Dev/architecture v10 category aggregator

### PR DESCRIPTION
**Pull request by TonyKatkov89**
Added category aggregator and adapter.

### CODE REVIEW

File: hypex/comparators/comparators.py [hunk 1]
```diff
@@ -68,4 +68,4 @@ def _inner_function(
 
     @staticmethod
     def _to_dataset(data: float, **kwargs) -> Dataset:
-        return Adapter.float_to_dataset(name="ATE", data=data)
+        return Adapter.value_to_dataset(data=data, col_name="ATE")
```
Comment (VasilievaKA) [inline hypex/comparators/comparators.py:71]: `col_name` to `column_name`?

File: hypex/dataset/abstract.py [hunk 2]
```diff
@@ -161,3 +161,6 @@ def to_json(self, filename: Optional[str] = None):
     @property
     def backend(self):
         return self._backend
+
+    def get_values(self, row: FieldKeyTypes = None, column: FieldKeyTypes = None) -> Any:
```
Comment (VasilievaKA) [inline hypex/dataset/abstract.py:165]: `Optional[FieldKeyTypes]`

File: hypex/dataset/backends/abstract.py [hunk 3]
```diff
@@ -245,6 +245,9 @@ def agg(self, func) -> Any:
         raise AbstractMethodError
 
     @abstractmethod
+    def get_values(self, row: FieldKeyTypes = None, column: FieldKeyTypes = None) -> Any:
+        raise AbstractMethodError
```
Comment (VasilievaKA) [inline hypex/dataset/backends/abstract.py:249]: `Optional[FieldKeyTypes]`

File: hypex/dataset/backends/abstract.py [hunk 4]
```diff
@@ -342,6 +345,9 @@ def merge(
     def drop(self, labels: FieldKeyTypes = "", axis: int = 1) -> Any:
```
Comment (VasilievaKA) [inline hypex/dataset/backends/abstract.py:345]: `Optional[FieldKeyTypes]` and "" to None

File: hypex/dataset/backends/abstract.py [hunk 5]
```diff
@@ -342,6 +345,9 @@ def merge(
...
         raise AbstractMethodError
 
+    def filter(self, items: Optional[List] = None, like: Optional[str] = None, regex: Optional[str] = None, axis: Optional[int] = None) -> Any:
+        return AbstractMethodError
```
Comment (VasilievaKA) [inline hypex/dataset/backends/abstract.py:349]: axis to default value 0

File: hypex/dataset/backends/pandas_backend.py [hunk 6]
```diff
@@ -189,8 +192,16 @@ def _get_column_type(self, column_name: str) -> str:
 
     # try try-except if necessary
     def _update_column_type(self, column_name: str, type_name: str):
-        if self.data[column_name].isna().sum() == 0:
-            self.data.loc[:, column_name] = self.data[column_name].astype(type_name)
+        try:
+            if self.data[column_name].isna().sum() == 0:
+                self.data.loc[:, column_name] = self.data[column_name].astype(type_name)
+        except Exception as e:
```
Comment (VasilievaKA) [inline hypex/dataset/backends/pandas_backend.py:198]: delete try-except

File: hypex/dataset/backends/pandas_backend.py [hunk 7]
```diff
@@ -240,8 +251,23 @@ class PandasDataset(PandasNavigation, DatasetBackendCalc):
     def __init__(self, data: Union[pd.DataFrame, Dict, str, pd.Series] = None):
         super().__init__(data)
 
+    def get_values(self, row: FieldKeyTypes = None, column: FieldKeyTypes = None) -> Any:
```
Comment (VasilievaKA) [inline hypex/dataset/backends/pandas_backend.py:254]: `Optional[FieldKeyTypes]`

File: hypex/dataset/backends/pandas_backend.py [hunk 8]
```diff
@@ -405,5 +431,17 @@ def merge(
     def drop(self, labels: FieldKeyTypes = "", axis: int = 1) -> pd.DataFrame:
         return self.data.drop(labels=labels, axis=axis)
 
-    def rename(self, columns: Dict[str, str]):
+    def filter(self, items: Optional[List] = None, like: Optional[str] = None, regex: Optional[str] = None, axis: Optional[int] = None) -> pd.DataFrame:
```
Comment (VasilievaKA) [inline hypex/dataset/backends/pandas_backend.py:434]: axis = 0

File: hypex/dataset/backends/pandas_backend.py [hunk 9]
```diff
@@ -405,5 +431,17 @@ def merge(
...
+        return self.data.filter(items=items, like=like, regex=regex, axis=axis)
+    
+    def rename(self, columns: Dict[str, str]):
```
Comment (VasilievaKA) [inline hypex/dataset/backends/pandas_backend.py:437]: -> pd.DataFrame

File: hypex/dataset/dataset.py [hunk 10]
```diff
@@ -505,6 +511,16 @@ def drop(self, labels: Any = None, axis: int = 1):
         )
         return Dataset(roles=t_roles, data=t_data)
 
+    def filter(
```
Comment (VasilievaKA) [inline hypex/dataset/dataset.py:514]: -> "Dataset"

File: hypex/dataset/dataset.py [hunk 11]
```diff
@@ -524,11 +540,23 @@ def shuffle(self, random_state: Optional[int] = None) -> "Dataset":
...
-            names.get(column, column): role for column, role in self.roles.items()
-        }
+        roles = {names.get(column, column): role for column, role in self.roles.items()}
         return Dataset(roles, data=self.backend.rename(names))
 
+    def replace(
+        self,
+        to_replace: Any = None,
+        value: Any = None,
+        inplace: bool = False,
```
Comment (VasilievaKA) [inline hypex/dataset/dataset.py:550]: delete inplace

File: hypex/experiments/aa.py [hunk 12]
```diff
@@ -5,9 +5,17 @@
 from hypex.experiments import Experiment, OnRoleExperiment
 from hypex.splitters import AASplitter
 from hypex.utils import SpaceEnum
+from hypex.transformers.filters import CVFilter, ConstFilter, NanFilter, CorrFilter, OutliersFilter
+from hypex.transformers.category_agg import CategoryAggregator
 
 AA_TEST = Experiment(
     executors=[
+        # CVFilter(),
```
Comment (VasilievaKA) [inline hypex/experiments/aa.py:13]: delete this file

File: hypex/transformers/__init__.py [hunk 13]
```diff
@@ -1,3 +1,5 @@
 from .shuffle import Shuffle
+from .filters import CVFilter, ConstFilter, NanFilter, CorrFilter, OutliersFilter
+from .category_agg import CategoryAggregator
 
-__all__ = ["Shuffle"]
+__all__ = ["Shuffle", "CVFilter", "ConstFilter", "NanFilter", "CorrFilter", "OutliersFilter", "CategoryAggregator"]
```
Comment (VasilievaKA) [inline hypex/transformers/__init__.py:5]: 😍🦄

File: hypex/transformers/abstract.py [hunk 14]
```diff
@@ -1,22 +1,29 @@
...
 
+    def execute(self, data: ExperimentData) -> ExperimentData:
+        data = data.copy(data=self.calc(data=data.ds))
+        return data
+
     @staticmethod
-    @abstractmethod
-    def _inner_function(data: Dataset, **kwargs) -> Any:
-        raise AbstractMethodError
+    def _list_unification(roles: Any):
```
Comment (VasilievaKA) [inline hypex/transformers/abstract.py:28]: delete

File: hypex/utils/adapter.py [hunk 15]
```diff
@@ -1,21 +1,45 @@
...
+        Returns:
+        Dataset: A Dataset object generated from the input data.
+        Raises:
+        ValueError: If the data type is not supported.
+        """
+        # Convert data based on its type
+        if isinstance(data, dict):
+            return Adapter.dict_to_dataset(data)
+        elif isinstance(data, list):
+            if not isinstance(col_name, list):
```
Comment (VasilievaKA) [inline hypex/utils/adapter.py:27]: list -> List

File: hypex/utils/adapter.py [hunk 16]
```diff
@@ -1,21 +1,45 @@
...
 
 from hypex.dataset import Dataset, InfoRole
+from hypex.utils import ScalarType
 
 
 class Adapter:
 
-    # def to_dataset(name: str, data: Union[float, int]) -> Dataset:
+    @staticmethod
+    def to_dataset(data: Any, col_name: Union[str, List]) -> Dataset:
```
Comment (VasilievaKA) [inline hypex/utils/adapter.py:12]: List[FieldKeyTypes] 
col_name -> column_name

**Review by VasilievaKA — COMMENTED** [review]
🗽🤫

File: hypex/comparators/comparators.py (hunk 1 above)
Comment (TonyKatkov89) [reply hypex/comparators/comparators.py:71]: fixed

File: hypex/dataset/abstract.py (hunk 2 above)
Comment (TonyKatkov89) [reply hypex/dataset/abstract.py:165]: fixed

File: hypex/dataset/backends/abstract.py (hunk 3 above)
Comment (TonyKatkov89) [reply hypex/dataset/backends/abstract.py:249]: fixed

File: hypex/dataset/backends/abstract.py (hunk 4 above)
Comment (TonyKatkov89) [reply hypex/dataset/backends/abstract.py:345]: fixed

File: hypex/dataset/backends/abstract.py (hunk 5 above)
Comment (TonyKatkov89) [reply hypex/dataset/backends/abstract.py:349]: fixed

File: hypex/dataset/backends/pandas_backend.py (hunk 6 above)
Comment (TonyKatkov89) [reply hypex/dataset/backends/pandas_backend.py:198]: fixed

File: hypex/dataset/backends/pandas_backend.py (hunk 7 above)
Comment (TonyKatkov89) [reply hypex/dataset/backends/pandas_backend.py:254]: fixed

File: hypex/dataset/backends/pandas_backend.py (hunk 9 above)
Comment (TonyKatkov89) [reply hypex/dataset/backends/pandas_backend.py:437]: fixed

File: hypex/dataset/dataset.py (hunk 10 above)
Comment (TonyKatkov89) [reply hypex/dataset/dataset.py:514]: fixed

File: hypex/dataset/dataset.py (hunk 11 above)
Comment (TonyKatkov89) [reply hypex/dataset/dataset.py:550]: fixed

File: hypex/experiments/aa.py (hunk 12 above)
Comment (TonyKatkov89) [reply hypex/experiments/aa.py:13]: during merge

File: hypex/transformers/abstract.py (hunk 14 above)
Comment (TonyKatkov89) [reply hypex/transformers/abstract.py:28]: fixed

File: hypex/utils/adapter.py (hunk 15 above)
Comment (TonyKatkov89) [reply hypex/utils/adapter.py:27]: fixed

File: hypex/utils/adapter.py (hunk 16 above)
Comment (TonyKatkov89) [reply hypex/utils/adapter.py:12]: fixd

**Pull request merged**